### PR TITLE
All Pico services should default to a lesser than DEFAULT_WEIGHT

### PR DESCRIPTION
**Pull request by trentjeff**
Fix for [issue#6583.
](https://github.com/helidon-io/helidon/issues/6583)

### CODE REVIEW

File: pico/api/src/main/java/io/helidon/pico/api/ServiceInfoBasics.java [hunk 1]
```diff
@@ -32,10 +32,11 @@
...
 
     /**
-     * Default weight for any weighted component (whether it implements this interface
-     * or uses {@link  io.helidon.common.Weight} annotation).
+     * Default weight for any <i>internal</i> Pico service component. It is defined to be
+     * {@link Weighted#DEFAULT_WEIGHT} {@code - 1} in order to allow any other service implementation to
+     * naturally have a higher weight (since it will use the {@code DEFAULT_WEIGHT} unless explicitly overridden.
      */
-    double DEFAULT_WEIGHT = Weighted.DEFAULT_WEIGHT;
+    double DEFAULT_PICO_WEIGHT = Weighted.DEFAULT_WEIGHT;
```
Comment (tomas-langer) [inline pico/api/src/main/java/io/helidon/pico/api/ServiceInfoBasics.java:39]: I do not see the `-1` mentioned in the javadoc

File: pico/configdriven/runtime/src/main/java/io/helidon/pico/configdriven/runtime/DefaultConfigBeanBuilderValidatorProvider.java [hunk 2]
```diff
@@ -20,12 +20,11 @@
 import io.helidon.builder.config.spi.ConfigBeanBuilderValidatorProvider;
 import io.helidon.common.LazyValue;
 import io.helidon.common.Weight;
-import io.helidon.common.Weighted;
 
 /**
  * Service-loaded provider for {@link io.helidon.builder.config.spi.ConfigResolverProvider}.
  */
-@Weight(Weighted.DEFAULT_WEIGHT)
+@Weight(io.helidon.pico.api.ServiceInfoBasics.DEFAULT_PICO_WEIGHT)
```
Comment (tomas-langer) [inline pico/configdriven/runtime/src/main/java/io/helidon/pico/configdriven/runtime/DefaultConfigBeanBuilderValidatorProvider.java:27]: Why is it fully qualified? There should not be a naming conflict.

Comment (tomas-langer) [reply pico/configdriven/runtime/src/main/java/io/helidon/pico/configdriven/runtime/DefaultConfigBeanBuilderValidatorProvider.java:27]: Seems to be in all usages

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: pico/api/src/main/java/io/helidon/pico/api/ServiceInfoBasics.java (hunk 1 above)
Comment (trentjeff) [reply pico/api/src/main/java/io/helidon/pico/api/ServiceInfoBasics.java:39]: Great catch! lol

File: pico/configdriven/runtime/src/main/java/io/helidon/pico/configdriven/runtime/DefaultConfigBeanBuilderValidatorProvider.java (hunk 2 above)
Comment (trentjeff) [reply pico/configdriven/runtime/src/main/java/io/helidon/pico/configdriven/runtime/DefaultConfigBeanBuilderValidatorProvider.java:27]: Yes, it was just easier to do it that way. Is that somehow a problem?  I did the same for the javadoc.

File: pico/api/src/main/java/io/helidon/pico/api/ServiceInfoBasics.java (hunk 1 above)
Comment (trentjeff) [reply pico/api/src/main/java/io/helidon/pico/api/ServiceInfoBasics.java]: pushed

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: pico/configdriven/runtime/src/main/java/io/helidon/pico/configdriven/runtime/DefaultConfigBeanBuilderValidatorProvider.java (hunk 2 above)
Comment (trentjeff) [reply pico/configdriven/runtime/src/main/java/io/helidon/pico/configdriven/runtime/DefaultConfigBeanBuilderValidatorProvider.java]: pushed

**Review by tomas-langer — APPROVED** [review]

**Pull request merged**